daft/functions/ai/__init__.py

```diff
@@ -73,7 +69,7 @@ def embed_text(
     Args:
         text (String Expression):
             The input text column expression.
-        provider (str | Provider | None):
+        provider (ProviderType | Provider | None):
```

Comment:
```suggestion
        provider (str | Provider | None):
```

You will need to update all usage of ProviderType back to `str` since it's an identifier not an enumeration. The ProviderType Literal is still useful for the `load_provider` first argument and for the key of the PROVIDERS dict for the various load_* impls.